datasets/Turkish_NER/Turkish_NER.py

```diff
@@ -0,0 +1,196 @@
+    """TODO: Short description of my dataset."""
+
+
+    def _info(self):
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=datasets.Features(
+                    {
+
+                    "id": datasets.Value("string"),
+                    "tokens": datasets.Sequence(datasets.Value("string")),
+                    "domains": datasets.Sequence(
+                        datasets.features.ClassLabel(
+                            names=[
+                            "location",
+                            "geography",
+                            "government",
+                            "law",
+                            "soccer",
+                            "sports",
+                            "military",
+                            "film", 
+                            "royalty",
+                            "book",
+                            "people",
+                            "music",
+                            "business",
+                            "religion",
+                            "time",
+                            "tv",
+                            "organization",
+                            "education"
+
+                            ])),
+                    "ner_tags": datasets.Sequence(
+                        datasets.features.ClassLabel(
+                            names=[
+                                "O",
+                                "B-PERSON",
+                                "I-PERSON",
+                                "B-ORGANIZATION",
+                                "I-ORGANIZATION",
+                                "B-LOCATION",
+                                "I-LOCATION",
+                                "B-MISC",
+                                "I-MISC", 
+                            ]
+                        )
+                    )                       
+                }
+
+            ),
+                        
+            supervised_keys=None,
+            # Homepage of the dataset for documentation
+            homepage=_HOMEPAGE,
+            # License for the dataset if available
+            license=_LICENSE,
+            # Citation for the dataset
+            citation=_CITATION,
+        )
+        
+        
+
+    def _split_generators(self, dl_manager):
+        """Returns SplitGenerators."""
+        global data_dir
+        data_dir = dl_manager.download_and_extract(_URL)
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                gen_kwargs={
+                    "filepath": os.path.join(data_dir, _FILE_NAME_ZIP),
+                    "split": "train",
+                },
+            ),
+        ]
+
+    def _generate_examples(self, filepath, split):
+        """ Yields examples. """
+        global data_dir
+        logging.info("⏳ Generating examples from = %s", filepath)
+        os.system(f"""unzip "{filepath}" -d {data_dir}""")
```

Comment:
I suggest you use `dl_manager.extract` function instead of os.system calls to get the zips inside the zip. (in the `_split_generators` function)

Reply:
I just did 👍 But this still didn't extract the file as a .DUMP. when I run load_dataset I get a .lock file of the dataset in the "extracted" in folder in cache, and I still get .arrow file in the dataset's folder (in cache) instead of the .DUMP. So when I tried to generate a dummy file with the script you've sent it can't find any matches for it (obviously). I don't know in which step it decides to unzip it that way or if it's unsuccessful in unzipping, tried unit tests but couldn't find. Thanks for the help though.

Reply:
try clearing the cache folder and lock files then try again, I had a similar problem before.